tools/Azure.Mcp.Tools.Cosmos/src/Services/CosmosService.cs

```diff
@@ -24,6 +25,35 @@ public class CosmosService(ISubscriptionService subscriptionService, ITenantServ
     private static readonly TimeSpan s_cacheDurationResources = TimeSpan.FromMinutes(15);
     private bool _disposed;
 
+    // Maximum number of results to prevent DoS attacks and performance issues
+    private const int MaxResultLimit = 1000;
+
+    // Maximum query length to prevent DoS attacks
+    private const int MaxQueryLength = 8000;
+
+    // Regex timeout to prevent ReDoS attacks
+    private static readonly TimeSpan RegexTimeout = TimeSpan.FromSeconds(3);
+
+    // Static arrays for security validation - initialized once per class
+    private static readonly string[] DangerousKeywords =
```

Comment:
Same comment as the other PR, should we move to an allow list instead of a disallow list?  Seems safer.

Reply:
Yes - I would agree. A allow whitelist would help to control the workflow much better.